vetapp/app/controllers/visits_controller.rb

```diff
@@ -0,0 +1,53 @@
+class VisitsController < ApplicationController
+       before_action :authenticate_user!
+
+    def index
+        @animal=Animal.all
+        @visit=Visit.all
+        @user=User.all
+    end
+    
+    def show
+    end
+    
+    def new
+        @visit=Visit.new
+        @animal=Animal.all
+        @user=User.all
+    end
+    
+    def create
+        @animal=Animal.all
+        @user=User.all
+        @visit=Visit.new
+        @visit.weight=params[:visit][:weight]
+        @visit.animal_id=params[:animal_id]
+        @visit.vet=params[:visit][:vet]
+        @visit.user_id=current_user.id
+        @visit.save!
+        redirect_to animal_path(params[:animal_id])
+    end
+    
+    def edit
+        @visit=Visit.find(params[:id])
+        @animal=Animal.all
+        @user=User.all
+    end
+    
+    def update
+        @animal=Animal.all
+        @visit=Visit.find(params[:id])
+        @visit.weight=params[:visit][:weight]
+        @visit.animal_id=params[:animal_id]
+        @visit.vet=params[:visit][:vet]
+        @visit.user_id=current_user.id
+        @visit.save!
```

Comment:
Same thing as with the `#create` action, you can whitelist attributes that you want to assign to your object to get around strong parameters.